client/client_test.go

```diff
@@ -0,0 +1,94 @@
+package client
+
+import (
+	"testing"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"github.com/heroku/silvia-runtime-university/spec"
+	"reflect"
+	"fmt"
+	"strings"
+)
+
+var testCases = []struct{
+	name string
+	inputPoints []spec.Point
+	wantFeatures []spec.Feature
+}{
+	{
+		name:         "With zero points",
+		inputPoints:  []spec.Point{},
+		wantFeatures: []spec.Feature{},
+	},
+	{
+		name:        "With one point",
+		inputPoints: []spec.Point{{ Latitude: 100001, Longitude: 100002},
+		},
+		wantFeatures: []spec.Feature{
+			{ Name: "somefeature", Location: &spec.Point{Latitude: 100001, Longitude: 100002} },
+		},
+	},
+	{
+		name:        "With two points",
+		inputPoints: []spec.Point{{ Latitude: 100001, Longitude: 100002}, {Latitude: 100003, Longitude: 100004 }},
+		wantFeatures: []spec.Feature{
+			{ Name: "somefeature", Location: &spec.Point{Latitude: 100001, Longitude: 100002} },
+			{ Name: "anotherfeature", Location: &spec.Point{Latitude: 100003, Longitude: 100004} },
+		},
+	},
+	{
+		name:        "With point unassociated to feature",
+		inputPoints: []spec.Point{{ Latitude: 100008, Longitude: 100009}},
+		wantFeatures: []spec.Feature{
```

Comment:
One other thought I had for this case would be to have something like `wantErr bool` in the test case struct to signal that you expect an error when you make this call. That would allow the tests below to check that and be a bit more explicit about the `No feature associated with this point` case -- I think right now if your fake client always returned that error no matter what your tests would still pass.

Reply:
I agree, and I like your solution. Updated!